src/engine/JoinHelpers.h

```diff
@@ -42,13 +46,18 @@ CPP_template(typename Input)(
     requires ad_utility::SameAsAny<Input, Result::Generator,
                                    Result::LazyResult>) LazyInputView
     convertGenerator(Input gen, OptionalPermutation permutation = {}) {
-  for (auto& [table, localVocab] : gen) {
+  auto transformer = [permutation = std::move(permutation)](auto& element) {
+    auto& [table, localVocab] = element;
     applyPermutation(table, permutation);
     // Make sure to actually move the table into the wrapper so that the tables
     // live as long as the wrapper.
-    ad_utility::IdTableAndFirstCol t{std::move(table), std::move(localVocab)};
-    co_yield t;
-  }
+    return ad_utility::IdTableAndFirstCol{std::move(table),
+                                          std::move(localVocab)};
+  };
+  auto cachingRange = ad_utility::CachingTransformInputRange(
+      std::move(gen), std::move(transformer));
+  return ad_utility::InputRangeTypeErased<
+      ad_utility::IdTableAndFirstCol<IdTable>>(std::move(cachingRange));
```

Comment:
```suggestion
 using namespace ad_utility;  // can be also further up for the `IdTableAndFirstCol`.
  return InputRangeTypeErased{CachingTransformInputRange(
      std::move(gen), std::move(transformer))};

```
(No need for the temporary.